go-controller/pkg/ovn/port_cache.go

```diff
@@ -0,0 +1,86 @@
+package ovn
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"time"
+)
+
+type portCache struct {
+	sync.Mutex
+
+	cache map[string]*lpInfo
+}
+
+type lpInfo struct {
+	name          string
+	uuid          string
+	logicalSwitch string
+	ip            *net.IPNet
+	mac           net.HardwareAddr
+
+	// removeTime marks the time after which this port should be removed
+	// from the cache
+	removeTime time.Time
+}
+
+func newPortCache(stopChan <-chan struct{}) *portCache {
+	c := &portCache{
+		cache: make(map[string]*lpInfo),
+	}
```

Comment:
@girishmg no because it's embedded into the cache struct. Also we don't actually need to initialize non-pointer mutexes as long as we don't pass them around. We should really make the existing mutexes consistent since we have a mix of pointer/non-pointer ones in the Controller struct.